chisel/src/solidity_helper.rs

```diff
@@ -60,7 +60,6 @@ impl SolidityHelper {
         let mut comments = Vec::with_capacity(DEFAULT_COMMENTS);
         let mut errors = Vec::with_capacity(5);
         let mut out = Lexer::new(input, 0, &mut comments, &mut errors)
-            .flatten()
             .map(|(start, token, end)| (start, token.style(), end))
```

Comment:
We don't need `flatten`. 

In `solang-parser`, the implementation for `Iterator` trait changes from ([0.3.0](https://github.com/hyperledger/solang/blob/53172f3ad3627b5281086c1ee6f13a57a512f346/solang-parser/src/lexer.rs#L744)): 

```rust
fn next(&mut self) -> Option<Result<'input>>
```
to ([0.3.1](https://github.com/hyperledger/solang/blob/v0.3.1/solang-parser/src/lexer.rs#L745)): 

```rust
fn next(&mut self) -> Option<Spanned<'input>> {
```